fem/geom.cpp

```diff
@@ -274,8 +274,9 @@ const IntegrationRule * Geometry::GetVertices(int GeomType)
       case Geometry::CUBE:        return GeomVert[5];
       case Geometry::PRISM:       return GeomVert[6];
       case Geometry::PYRAMID:     return GeomVert[7];
-      default:
-         mfem_error ("Geometry::GetVertices(...)");
+      case Geometry::INVALID:
```

Comment:
Just wanted to quickly reply here. The value to removing `default:` from the switch statements over enum is that you will get a compiler warning for cases not handled. This improves safety in the long term for when options are added (so that adding an option to the enum will light up where the developer may not have accounted for side effects in the code). This also applies to https://github.com/mfem/mfem/pull/3846#discussion_r1397922825. I believe compilers being able to warn on missing cases is a recent thing but this has become favorable practice (https://twitter.com/bobrippling/status/1157039794809135109).

Reply:
I appreciate this. However, should this be in this PR here or should we make a separate PR where we also enforce the warning flag for builds, at least for CI?

Reply:
I don't think it's a change we need to go an make library wide, but for now if we're touching these parts of the code, we might as well change it. This flag is already tested as part of the CI with the `-Werror -Wall` builds (see https://github.com/mfem/github-actions/blob/ff5143d4ef55622afdacc945960b097ab790ef8c/build-mfem/action.yml#L91C11-L91C48).